govc/dvs/create.go

```diff
@@ -94,6 +97,14 @@ func (cmd *create) Run(ctx context.Context, f *flag.FlagSet) error {
 		}
 	}
 
+	numUplinkPorts := int(cmd.numUplinkPorts)
+
+	var policy types.DVSNameArrayUplinkPortPolicy
+	for i := 0; i < numUplinkPorts; i++ {
+		policy.UplinkPortName = append(policy.UplinkPortName, fmt.Sprintf("Uplink %d", i+1))
+	}
+	cmd.configSpec.UplinkPortPolicy = &policy
```

Comment:
Should `UplinkPortPolicy` only be set `if numUplinkPorts > 0 {}` ?

Reply:
Thanks for calling that out - adding num-uplinks=0 fails with the following error message.  vSphere 7.0u3 won't let you create zero uplinks (stating min of 1) in the GUI either.  
```
go/bin/govc: ServerFaultCode: 
Required property uplinkPortName is missing from data object of type DVSNameArrayUplinkPortPolicy

while parsing serialized DataObject of type vim.DistributedVirtualSwitch.NameArrayUplinkPortPolicy
at line 2, column 304

while parsing property "uplinkPortPolicy" of static type DVSUplinkPortPolicy

while parsing serialized DataObject of type vim.dvs.VmwareDistributedVirtualSwitch.ConfigSpec
at line 2, column 143

while parsing property "configSpec" of static type DVSConfigSpec

while parsing serialized DataObject of type vim.DistributedVirtualSwitch.CreateSpec
at line 2, column 137

while parsing call information for method CreateDVS_Task
at line 2, column 66

while parsing SOAP body
at line 2, column 60

while parsing SOAP envelope
at line 2, column 0

while parsing HTTP request for method createDistributedVirtualSwitch
on object of type vim.Folder
at line 1, column 0
```
I was checking for more than 1 uplink in the old PR: I will add that back in as I don't think the above output is really helpful in telling the user what the problem is if that's ok? we were discussing it in the old PR here:  
https://github.com/vmware/govmomi/pull/2499/commits/9e96c0220e356707ad5de7ea3ef6977d99ef9408

I also don't like the idea of the help showing a default (zero) that's invalid, so can I add back in the line from my old PR. 
	cmd.numUplinkPorts = 2 // Set Default Uplink ports to 2

Love to hear your thoughts before making the code changes.
Thanks in advance, 
H.

Reply:
I'd rather not change the default behavior. NumUplinkPorts has no default value in [PowerCLI](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/cd82421a-ee01-40a9-97c3-424ae49db692/972fed92-0e32-4993-9d35-0557c829bebd/doc/New-VDSwitch.html) either and is not a required param.  I think it'd be best to add the uplinks > 0 check back and default to '0'.  Seems ok to me if help shows '0' as the default, you can also clarify in the flag help or the command's `Description()` function.

I'm not sure why the GUI requires at least 1, last time I used to the GUI for this (while developing the govc command) was 6+ years ago.

Reply:
Sounds good.